src/switchroot/ostree-mount-util.h

```diff
@@ -106,6 +120,69 @@ read_proc_cmdline_ostree (void)
   return ret;
 }
 
+static inline char*
+gen_ostree_target (const char* prepend)
+{
+  char dir_str[PATH_MAX];
+  int dir_str_len = snprintf(dir_str, sizeof(dir_str), "%s", prepend);
+  int dir_str_cap = PATH_MAX - dir_str_len;
+  int prepend_len = dir_str_len;
+  dir_str_len += snprintf(dir_str + prepend_len, sizeof(dir_str), "%s", "/ostree");
+  dir_str_cap = PATH_MAX - dir_str_len;
+  char* append_from = dir_str + dir_str_len;
+  int appended_len = 0;
+  DIR __attribute__ ((cleanup(close_dir))) *dir = opendir(dir_str);
+  if (!dir)
+    {
+      fprintf(stderr, "'%s' %d", strerror(errno), __LINE__);
+      return "";
+    }
+
+  struct dirent *ent = 0;
+  for (time_t latest = 0; (ent = readdir(dir)); )
+    {
+      if (ent->d_type == DT_LNK)
```

Comment:
Usually I try to avoid nesting by doing e.g.:
```
  if (ent->d_type != DT_LNK)
    continue;
```

Reply:
Makes sense